src/cli/test/test_with_analysis.py

```diff
@@ -80,5 +97,100 @@ def test_run_with_analysis(osclipath, is_labs: bool):
+    run_log = run_log_path.read_text()
+    assert "No results for objective function IsNonExisting.NonExisting" in run_log
+    assert "runner.registerWarning called" in run_log
+    assert len(run_log.splitlines()) <= 3
+
+    assert count_msg_with_log_level(run_log=run_log, log_level=LogLevel.Trace, is_labs=is_labs) == 0
+    assert count_msg_with_log_level(run_log=run_log, log_level=LogLevel.Debug, is_labs=is_labs) == 0
+    assert count_msg_with_log_level(run_log=run_log, log_level=LogLevel.Info, is_labs=is_labs) == 0
+    assert 1 < count_msg_with_log_level(run_log=run_log, log_level=LogLevel.Warn, is_labs=is_labs) < 4
+    assert count_msg_with_log_level(run_log=run_log, log_level=LogLevel.Error, is_labs=is_labs) == 0
+    assert count_msg_with_log_level(run_log=run_log, log_level=LogLevel.Fatal, is_labs=is_labs) == 0
+
+
+@pytest.mark.parametrize(
+    "is_labs",
+    [pytest.param(True, id="labs"), pytest.param(False, id="classic")],
+)
+def test_run_log_debug(osclipath, is_labs: bool):
+    suffix = "labs" if is_labs else "classic"
+    suffix += "_debug"
+    runDir = run_workflow(osclipath=osclipath, suffix=suffix, is_labs=is_labs, verbose=False, debug=True)
+    assert runDir.is_dir()
+    run_log_path = runDir / "run.log"
+    assert run_log_path.is_file()
+    run_log = run_log_path.read_text()
+    assert "No results for objective function IsNonExisting.NonExisting" in run_log
+    assert "runner.registerWarning called" in run_log
+    assert 50 < len(run_log.splitlines()) < 100
+
+    assert count_msg_with_log_level(run_log=run_log, log_level=LogLevel.Trace, is_labs=is_labs) == 0
+    assert 30 < count_msg_with_log_level(run_log=run_log, log_level=LogLevel.Debug, is_labs=is_labs) < 40
+    assert 20 < count_msg_with_log_level(run_log=run_log, log_level=LogLevel.Info, is_labs=is_labs) < 50
+    assert 1 < count_msg_with_log_level(run_log=run_log, log_level=LogLevel.Warn, is_labs=is_labs) < 4
+    assert count_msg_with_log_level(run_log=run_log, log_level=LogLevel.Error, is_labs=is_labs) == 0
+    assert count_msg_with_log_level(run_log=run_log, log_level=LogLevel.Fatal, is_labs=is_labs) == 0
+
+
+@pytest.mark.parametrize(
+    "is_labs",
+    [pytest.param(True, id="labs"), pytest.param(False, id="classic")],
+)
+def test_run_log_toplevel_verbose(osclipath, is_labs: bool):
+    suffix = "labs" if is_labs else "classic"
+    suffix += "_toplevel_verbose"
+    runDir = run_workflow(osclipath=osclipath, suffix=suffix, is_labs=is_labs, verbose=True, debug=False)
+    assert runDir.is_dir()
+    run_log_path = runDir / "run.log"
+    assert run_log_path.is_file()
+    run_log = run_log_path.read_text()
+    assert "No results for objective function IsNonExisting.NonExisting" in run_log
+    assert "runner.registerWarning called" in run_log
+    if not is_labs:
+        pytest.skip("classic (Ruby) CLI just ignored the top level --verbose flag for run.log")
+    assert 50 < len(run_log.splitlines()) < 100
```

Comment:
If you did `openstudio --verbose run -w workflow.osw` , would you expect the `run.log` to contain **Debug** log messages?

As a reminder, we have ALL these ways of callling the CLI

```
openstudio --verbose ...
openstudio run --debug ...
openstudio run -w workflow.osw with workflow.osw having "run_options" > debug in there
# Labs only
openstudio --loglevel Debug ...
```


Follow up question: knowing that the Ruby CLI just didn’t set the run.log to Debug in that case (it only cares about run --debug ), would you:

* A) Use the historical behavior.
* B) Change it to respect the top level log level flag.
     * if `--debug` is passed, it will increase the log level to `Debug` if not already, but for `openstudio --loglevel Trace run --debug -w in.osw` , the Log Level for stdout and run.log would stay at `Trace`)

Reply:
If this is going to be merged into our imminent release, then I think we need to take option A and mirror the historical behavior (plus now the --loglevel option). 

However, I think post release (and early in the cycle) we should make a change to remove the logging option under the run subcommand.

@brianlball @shorowit @joseph-robertson